api/v2/views/base.py

```diff
@@ -24,9 +24,16 @@ def unresolved_requests_only(fn):
     @wraps(fn)
     def wrapper(self, request, *args, **kwargs):
         instance = self.get_object()
-        user_can_act = request.user.is_staff or request.user.is_superuser or CloudAdministrator.objects.filter(user=request.user.id).exists()
-        #TODO: Logic needs 're-worked' here. MachineRequests in 'non-final' states should be allowed to be PATCH'ed for re-submission.
-        if (not user_can_act and hasattr(instance, "is_closed") and instance.is_closed()):
+        staff_can_act = request.user.is_staff or request.user.is_superuser or CloudAdministrator.objects.filter(user=request.user.id).exists()
+        user_can_act = request.user == getattr(instance, 'created_by')
+        if not user_can_act and not staff_can_act:
+            message = (
+                "Method '%s' not allowed: "
+                "Only staff members and the owner are authorized to make this request."
+                % self.request.method
+            )
+            raise exceptions.NotAuthenticated(detail=message)
+        if (not staff_can_act and hasattr(instance, "is_closed") and instance.is_closed()):
```

Comment:
Possibly modify to just test for `instance.is_closed()`.

Also write a comment to explain this logic in English.